timemachine/cpp/src/nonbonded_all_pairs.cu

```diff
@@ -31,19 +35,15 @@ NonbondedAllPairs<RealType>::NonbondedAllPairs(
                     &k_nonbonded_unified<RealType, 1, 1, 0>,
                     &k_nonbonded_unified<RealType, 1, 1, 1>}) {
 
-    if (atom_idxs) {
-        cudaSafeMalloc(&d_atom_idxs_, K_ * sizeof(*d_atom_idxs_));
-        std::vector<int> atom_idxs_v(atom_idxs->begin(), atom_idxs->end());
-        gpuErrchk(cudaMemcpy(d_atom_idxs_, &atom_idxs_v[0], K_ * sizeof(*d_atom_idxs_), cudaMemcpyHostToDevice));
-    }
+    cudaSafeMalloc(&d_atom_idxs_, N_ * sizeof(*d_atom_idxs_));
 
-    cudaSafeMalloc(&d_sorted_atom_idxs_, K_ * sizeof(*d_sorted_atom_idxs_));
+    cudaSafeMalloc(&d_sorted_atom_idxs_, N_ * sizeof(*d_sorted_atom_idxs_));
```

Comment:
(guessing it had to do with letting K_ be variable across different calls, but wasn't 100% sure)

Reply:
Correct, it is so that the nonbonded potential can use different values of K. Could also reconstruct a new NB potential in the local md, but this seemed cleaner

Reply:
Okay - wasn't sure if it was safe to malloc now of size `N_` instead of size `K_` - i.e. if we only initially the `K_` subsets of the resulting arrays, is it safe that the rest of the array is uninitialized?

Reply:
It is safe as long as we maintain our convention of passing `K_` to the kernels and avoid going beyond that. An example is: https://github.com/proteneer/timemachine/blob/master/timemachine/cpp/src/kernels/nonbonded_common.cu#L19-L21